src/main/java/org/mskcc/limsrest/limsapi/cmoinfo/converter/SampleToCorrectedCmoIdConverter.java

```diff
@@ -49,8 +52,33 @@ public CorrectedCmoSampleView convert(Sample sample) throws LimsException {
 
             return correctedCmoSampleView;
         } catch (Exception e) {
-            throw new RuntimeException(String.format("Error while retrieving information for sample: %s. Couse: %s",
-                    sample.getIgoId(), e.getMessage()));
+            throw new RuntimeException(String.format("Error for sample %s: %s",
+                    sample.getIgoId(), e.getMessage()), e);
         }
     }
+
+    private void validate(Sample sample) {
+        StringBuilder error = new StringBuilder();
+
+        error = appendToError(sample, s -> (SampleType) SampleType.fromString(sample.getExemplarSampleType()), error);
+        error = appendToError(sample, s -> (Recipe) Recipe.getRecipeByValue(sample.get(Sample.RECIPE)), error);
+        error = appendToError(sample, s -> (SampleClass) SampleClass.fromValue(sample.getCorrectedSampleClass()),
+                error);
+        error = appendToError(sample, s -> (SampleOrigin) SampleOrigin.fromValue(sample.getCorrectedCmoSampleOrigin()
+        ), error);
+        error = appendToError(sample, s -> (SpecimenType) SpecimenType.fromValue(sample.getCorrectedSpecimenType()),
+                error);
+
+        throw new RuntimeException(error.toString());
```

Comment:
if `error ` is empty, exception will be thrown anyway?

Reply:
Yep, I changed it and will commit changes